syntax = "proto3";

service EmployeeService{
  rpc GetEmployee (EmployeeRequest) returns (Employee){}
  rpc SetEmployee (Employee) returns (ResponseMessage){}
}

service DeptService{
  rpc GetDept (DeptRequest) returns (Dept){}
  rpc SetDept (Dept) returns (ResponseMessage){}
}


message Employee{
  int32 id = 1;
  string name  = 2;
  string designation = 3;
  float salary = 4;
  Dept department = 5;
}

message EmployeeRequest{
  int32 id = 1;
}

message Dept{
  int32 id = 1;
  string name = 2;
}

message DeptRequest{
  int32 id = 1;
}


message ResponseMessage{
  string msg = 1;
}
